packages/react-i18n/docs/react_i18n_schema.md

```diff
@@ -0,0 +1,295 @@
+
+## Serialization to JSON+Comments
+
+`JSON+Comments` is a serialization format whose syntax is a superset of `JSON`, but a subset of [`JSON5`](https://json5.org/).
+
+```
+JSON < JSON+Comments < JSON5
+```
+
+Historically, `JSON` had support for comments, but this [was then removed](https://archive.vn/20150704102718/https://plus.google.com/+DouglasCrockfordEsq/posts/RK8qyGVaGSr). However, comments have proven to be a useful mechanism for developers to [provide additional context to translators](https://development.shopify.io/engineering/developing_at_Shopify/internationalization/providing_context_notes). The `JSON+Comments` serialization format extends the `JSON` serialization to re-add support for comments.
+
+The `JSON+Comments` can be parsed using a `JSON5` parser, or a `JSON` parser that is not sensitive to comments.
+
+**Example React I18n JSON+Commments file**
+
+```json5
+{
+  // This is a context comment for `hello`!
+  "hello": "My name is {name}",
+  "parent": {
+    "child_1": "I am the first child of `parent`!", // This is a trailing context comment for `child_1`!
+    "child_2": {
+      /*
+        This is a multi-line
+        context comment for `grandchild_1`!
+      */
+      "grandchild_1": "I am the first grandchild of `parent`!",
+      "grandchild_2": "I am the second grandchild of `parent`!" /* This is a trailing context comment for `grandchild_2`! */
+    }
+  }
+}
+```
+
+Comments follow the [JSON5 syntax for comments](https://spec.json5.org/#comments): `//` signals the start of a comment that ends at the end of the line. `/*` signals the start of a comment that ends with the next `*/`.
+
+### Why not JSON5?
+
+Ideally, we would not have needed a `JSON+Comments` serialization, and instead just defined a [`JSON5`](https://json5.org/) serialization instead.
+However, `JSON5` parsers are immature, and notably there are no mature `JSON5` parsers in Ruby. Given that important consumers of React I18n files are written in Ruby, it would not make sense to define a serialization using the full syntax of `JSON5`, since those files would not be able to be handled.
+
+Adding a `JSON5` serialization would be an excellent future addition to React I18n, at a point in time where these consumers could be made to understand `JSON5`.
+
+Since `JSON+Comments` is a subset of `JSON5`, a `JSON5` parser can be used to parse `JSON+Comments` serialized files.
+
+# Comparison with Rails I18n
+
+React I18n's schema was loosely based on the Rails I18n schema.
+This section compares the two schemas, in order to point out the ways in which they differ from one another.
+
+### Root locale key
+
+Rails I18n has a locale root key, while React I18n doesn't:
+
+**Rails I18n:**
+```yaml
+en:
+  foo: bar
+```
+
+**React I18n**
+
+```json
+{
+  "foo": "bar"
+}
+```
+
+Note the `en` (for English) locale key that is the root key for the Rails I18n file. This key is not present in the React I18n file.
```

Comment:
```suggestion
Note the `en` (for English) locale key that is the root key for the Rails I18n file. This key is not present in the React I18n file because it is denoted by filename.
```